jsapp/js/constants.ts

```diff
@@ -218,46 +280,66 @@ export const ASSET_FILE_TYPES = {
+  calculate: {label: t('Calculate'), icon: 'qt-calculate', id: QuestionTypeName.calculate},
+  date: {label: t('Date'), icon: 'qt-date', id: QuestionTypeName.date},
+  datetime: {label: t('Date & time'), icon: 'qt-date-time', id: QuestionTypeName.datetime},
+  decimal: {label: t('Decimal'), icon: 'qt-decimal', id: QuestionTypeName.decimal},
+  'xml-external': {label: t('External XML'), icon: 'qt-external-xml', id: QuestionTypeName['xml-external']},
+  file: {label: t('File'), icon: 'qt-file', id: QuestionTypeName.file},
+  geopoint: {label: t('Point'), icon: 'qt-point', id: QuestionTypeName.geopoint},
+  geoshape: {label: t('Area'), icon: 'qt-area', id: QuestionTypeName.geoshape},
+  geotrace: {label: t('Line'), icon: 'qt-line', id: QuestionTypeName.geotrace},
+  hidden: {label: t('Hidden'), icon: 'qt-hidden', id: QuestionTypeName.hidden},
+  image: {label: t('Photo'), icon: 'qt-photo', id: QuestionTypeName.image},
+  integer: {label: t('Number'), icon: 'qt-number', id: QuestionTypeName.integer},
+  kobomatrix: {label: t('Question Matrix'), icon: 'qt-question-matrix', id: QuestionTypeName.kobomatrix},
+  note: {label: t('Note'), icon: 'qt-note', id: QuestionTypeName.note},
+  range: {label: t('Range'), icon: 'qt-range', id: QuestionTypeName.range},
+  rank: {label: t('Ranking'), icon: 'qt-ranking', id: QuestionTypeName.rank},
+  score: {label: t('Rating'), icon: 'qt-rating', id: QuestionTypeName.score},
+  select_multiple: {label: t('Select Many'), icon: 'qt-select-many', id: QuestionTypeName.select_multiple},
+  select_one: {label: t('Select One'), icon: 'qt-select-one', id: QuestionTypeName.select_one},
+  text: {label: t('Text'), icon: 'qt-text', id: QuestionTypeName.text},
+  time: {label: t('Time'), icon: 'qt-time', id: QuestionTypeName.time},
+  video: {label: t('Video'), icon: 'qt-video', id: QuestionTypeName.video},
 });
 
+/**
+ * These are the types of survey rows that users can create in FormBuilder (as
+ * checkboxes) and ones that have data submitted automatically.
+ */
+export enum MetaQuestionTypeName {
+  start = 'start',
+  end = 'end',
+  today = 'today',
+  username = 'username',
+  simserial = 'simserial',
+  subscriberid = 'subscriberid',
+  deviceid = 'deviceid',
+  phonenumber = 'phonenumber',
+  audit = 'audit',
+  'background-audio' = 'background-audio',
+}
 export const META_QUESTION_TYPES = createEnum([
-  'start',
-  'end',
-  'today',
-  'username',
-  'simserial',
-  'subscriberid',
-  'deviceid',
-  'phonenumber',
-  'audit',
-  'background-audio',
-]);
+  MetaQuestionTypeName.start,
+  MetaQuestionTypeName.end,
+  MetaQuestionTypeName.today,
+  MetaQuestionTypeName.username,
+  MetaQuestionTypeName.simserial,
+  MetaQuestionTypeName.subscriberid,
+  MetaQuestionTypeName.deviceid,
+  MetaQuestionTypeName.phonenumber,
+  MetaQuestionTypeName.audit,
+  MetaQuestionTypeName['background-audio'],
+]) as {[P in MetaQuestionTypeName]: MetaQuestionTypeName};
```

Comment:
Is this setting each attribute in `MetaQuestionTypeName` to the entire type? I'm a bit confused by this and `GroupTypeBeginName`

Reply:
I didn't want to get rid of all those `SOMETHING = createEnum()` right now, but ultimately they should become `enum`s.

`createEnum` function is returning `{[val: string]: string}` which means it's an object with string properties and values. Using `createEnum` as is doesn't allow for the created enum to be more meaningful. The `SOMETHING = createEnum(…) as SomeType` is a way of overriding the returned type. It's not 100% nice TypeScript, but it's good way of going half way.